statedb/treeview.go

```diff
@@ -58,11 +91,26 @@ func (v *TreeView) Get(key []byte) ([]byte, error) {
 	return v.tree.Get(nil, key)
 }
 
-// Iterate iterates over all nodes of this tree.
-func (v *TreeView) Iterate(callback func(key, value []byte) bool) error {
+// IterateNodes iterates over all nodes of this tree.  The key and value are
+// the Arbo database representation of a node, and don't match the key value
+// used in Get, Add and Set.  When callback returns true, the iteration is
+// stopped and this function returns.
+func (v *TreeView) IterateNodes(callback func(key, value []byte) bool) error {
 	return v.tree.Iterate(nil, callback)
 }
 
+// Iterate iterates over all leafs of this tree.  When callback returns true,
+// the iteration is stopped and this function returns.
+func (v *TreeView) Iterate(callback func(key, value []byte) bool) error {
```

Comment:
This method also is implemented in `TreeUpdate` (`statedb/treeupdate.go`) with the exact same code, maybe can be abstracted in a way that there is only one implementation used by both structs/interfaces?